ci/jwt/jwks_server.py

```diff
@@ -0,0 +1,130 @@
+                'ES512': jwk.JWK.generate(kty='EC', crv='P-521'),
+                'HS256': jwk.JWK.generate(kty='oct', size=256),
+                'HS384': jwk.JWK.generate(kty='oct', size=384),
+                'HS512': jwk.JWK.generate(kty='oct', size=512)
+            }.get(alg, None)
+        return keys[key_name]
+
+    def export_key(self, key):
+        try:
+            key_json = key.export(private_key=False)
+        except jwk.InvalidJWKType:
+            key_json = key.export()
+        key_dict = json.loads(key_json, object_pairs_hook=OrderedDict)
+        key_dict['kid'] = key.thumbprint()
+        return key_dict
+
+    def base64_padding(self, value):
+        return value + "=" * (-len(value) % 4)
+
+    def decode_token(self, token):
+        head_b64, payl_b64, sig = token.split(".", 3)
+        head = base64.urlsafe_b64decode(self.base64_padding(head_b64))
+        payl = base64.urlsafe_b64decode(self.base64_padding(payl_b64))
+        head_dict = json.loads(head, object_pairs_hook=OrderedDict)
+        payl_dict = json.loads(payl, object_pairs_hook=OrderedDict)
+        return head_dict, payl_dict
+
+    def do_GET(self): # pylint: disable=C0103
+        logging.info("GET: %s", str(self.path))
+        parts = self.path.strip('/').split('/')
+
+        if len(parts) == 2:
+            key_name, alg = parts[0], parts[1]
+            key = self.get_key(key_name, alg)
+            resp = json.dumps({
+                'keys': [
+                    self.export_key(key)
+                ]
+            })
+        else:
+            key_list = []
+            for key in keys.values():
+                key_list.append(self.export_key(key))
+            resp = json.dumps({
+                'keys': key_list
+            })
+
+        self.reply(resp)
+
+    def do_POST(self): # pylint: disable=C0103
+        logging.info("POST %s", str(self.path))
+
+        content_length = int(self.headers['Content-Length'])
+        post_data = self.rfile.read(content_length)
+        logging.info("BODY %s", post_data.decode('utf-8'))
+
+        key_name, alg = self.path.strip('/').split('/')
+        key = self.get_key(key_name, alg)
+
+        head, payl = self.decode_token(post_data.decode('utf-8'))
+        head['alg'] = alg
+        head['kid'] = key.thumbprint()
+
+        token = jwt.JWT(header=head, claims=payl)
+        token.make_signed_token(key)
+        self.reply(token.serialize())
+
+    def do_DELETE(self): # pylint: disable=C0103
+        logging.info("DELETE %s", str(self.path))
+        parts = self.path.strip('/').split('/')
+        if len(parts) == 2:
+            del keys[parts[0]]
+        else:
+            keys.clear()
+        self.reply("DELETED")
+
+
+def run(port, server_class=HTTPServer, handler_class=JWKSRequestHandler):
+    server_address = ('', port)
+    httpd = server_class(server_address, handler_class)
+    logging.info('Starting JWKS server...')
+    logging.info('Port: %d', port)
```

Comment:
Use fstring

Reply:
Done

Reply:
See there: https://stackoverflow.com/questions/34619790/pylint-message-logging-format-interpolation
There's recommendation to use `%` string format in logging functions.

Reply:
I don't understand from there why. But in my opinion all string formatting in same project should be in one way. And the most readble way in my opinion is fstring

Reply:
Because fstring resolves string in the line when it's written.
In opposite `%` will resolve string only if it need actually to log it, mostly depends log level.